distributed/scheduler.py

```diff
@@ -2688,10 +2704,14 @@ def transition_queued_processing(self, key: str, stimulus_id: str) -> RecsMsgs:
             assert not ts.actor, f"Actors can't be queued: {ts}"
             assert ts in self.queued
 
-        if ws := self.decide_worker_rootish_queuing_enabled():
-            self.queued.discard(ts)
-            worker_msgs = self._add_to_processing(ts, ws)
-        # If no worker, task just stays `queued`
+        if self.is_rootish(ts):
```

Comment:
This if/else gives me flashbacks to https://github.com/dask/distributed/pull/7280 / https://github.com/dask/distributed/pull/7221 / https://github.com/dask/distributed/pull/7262. Since we still have that inconsistency where `is_rootish` can change while the task is in the queue, I'd just want to think through the other possible reasons why `ts` might not be rootish here.